src/core/core.scale.js

```diff
@@ -1006,12 +1006,38 @@ class Scale extends Element {
 			tx2 = borderValue - axisHalfWidth;
 			x1 = alignBorderValue(chartArea.left) + axisHalfWidth;
 			x2 = chartArea.right;
-		} else {
+		} else if (position === 'right') {
 			borderValue = alignBorderValue(me.left);
 			x1 = chartArea.left;
 			x2 = alignBorderValue(chartArea.right) - axisHalfWidth;
 			tx1 = borderValue + axisHalfWidth;
 			tx2 = me.left + tl;
+		} else if (axis === 'x') {
```

Comment:
I think I'd rather put this logic in `layouts` and set a variable, `borderPixel` for example, to the scale (box). And use the variable here and in `_computeLabelItems`.

Reply:
or just set the top/bottom correctly and use those here as if this was a bottom scale.

Reply:
My initial thought was to use the layout system as well. The reason I didn't was that it would mean that the layout system would have to differentiate scales from other boxes. I wasn't sure if that was going to work long-term or not